core/commands/repo.go

```diff
@@ -65,24 +66,11 @@ order to reclaim hard disk space.
 
 		go func() {
 			defer close(outChan)
-			var errors []error
-			for res := range gcOutChan {
-				if res.KeyRemoved != nil {
-					outChan <- &corerepo.KeyRemoved{res.KeyRemoved}
-				}
-				if res.Error != nil {
-					errors = append(errors, res.Error)
-				}
-			}
-			switch len(errors) {
-			case 0:
-				return
-			case 1:
-				res.SetError(errors[0], cmds.ErrNormal)
-				return
-			default:
-				res.SetError(corerepo.NewMultiError(errors...), cmds.ErrNormal)
-				return
+			err := corerepo.CollectResult(gcOutChan, func(k *cid.Cid) {
```

Comment:
(discussed in IRC, putting here for posterity)

Any time youre sending or receiving on a channel that is outside the control of your immediate goroutine, its a good idea to use a context. In this case, the reader is the commands lib, and the commands lib will stop reading from the channel if the user cancels the operation.